lib/awful/screenshot.lua

```diff
@@ -0,0 +1,884 @@
+---------------------------------------------------------------------------
+--- Screenshots and related configuration settings
+--
+-- @author Brian Sobulefsky &lt;brian.sobulefsky@protonmail.com&gt;
+-- @copyright 2021 Brian Sobulefsky
+-- @inputmodule awful.screenshot
+---------------------------------------------------------------------------
+
+-- Grab environment we need
+local capi = { root = root,
+               screen = screen,
+               client = client,
+               mousegrabber = mousegrabber }
+local aw_screen = require("awful.screen")
+local gears = require("gears")
+local beautiful = require("beautiful")
+local wibox = require("wibox")
+local cairo = require("lgi").cairo
+local naughty = require("naughty")
+
+-- The module to be returned
+local module = { mt = {} }
+-- The screenshow object created by a call to the screenshow module
+local screenshot = {}
+-- The various methods of taking a screenshow (root window, client, etc).
+local screenshot_methods = {}
+
+-- Configuration data
+local ss_dir = nil
+local ss_prfx = nil
```

Comment:
but those would be nice also to make more maintainable

if you have troubles typing long names - you can just press Ctrl+xn in vim to auto-complete it

Reply:
Yeah its OK, I don't mind. If you look, most of the names are descriptive. It is usually just temporaries in helper functions where I don't do that, but I'll clean them up for the next push.